### DISCUSSION TITLE
クエストに対処が難しすぎるモンスターが出現する件の調整

### DISCUSSION THREAD

**Discussion by dis-** [hengband/hengband#1094] [Closed]
より自由に個性豊かなモンスターを実装するためには固定クエストとの噛み合いの悪さを解消しておく必要があると考えています。
範囲と具体的な実装については幾つか検討する余地があると思うので意見を募集します。
こちらの想定を幾つか出しておきます。

1. RES_ALL持ちがクエストに出現するとき、一定確率で再抽選する。または出現しないようにする。
2. 固定クエストへの出現率を下げるフラグを実装する。これを持つモンスターが固定クエストに出現するとき、一定確率で再抽選する。
3. クエストに出現する階層判定を上げるフラグを実装する。これを持つモンスターが固定クエストに出現するとき、モンスターレベル*1.5 > クエストレベル なら再抽選する。

**Comment by RRyukkuriRyu** [comment]
alpha23の新モンスターにもRES_ALL持ちがいますし、今後も増える可能性を考えると1の案を支持します。
固定クエストのランダムモンスター枠を無くすというのはどうでしょうか？固定モンスターの召喚によって攻略失敗するのは設計通りでしょうし対策手段もありますが、ランダム枠は完全に対策することが難しいので、いっそ無くして別の固定モンスターに差し替えた方が不確定要素が減って理不尽を感じることが少なくなると思います。何を配置するのかという議論は発生してしまいますが、出現モンスターが固定なら召喚先を絞れるので、調整が必要なモンスターを少なくすることができ、開発的にも楽になると思います。

**Reply by dis-** [reply]
ランダム枠の廃止は現行のクエスト全てに代替枠を設定しなければならないので、作業量が多いのが難点ですね。
今後クエストにランダム枠を導入し難くなり、自由度が低くなるという問題点もあります。
この2点を改善する妙案がなければ実現可能性が低いと返答させていただきます。

**Comment by dis-** [comment]
4. 固定クエストに出現するモンスターの生成判定ではレアリティを通常の二乗(上限100)で判定する

危険度が高い/理不尽度が高いモンスターは高レアリティに設定されがちなので、頻度を激減させることができそう。
クエストにだけあまり出てほしくないモンスターをレア3～レア5に設定するだけなので調整側も楽なのが良い。

**Comment by oguchi2918** [comment]
固定クエストでは召喚を含めて集団生成のモンスターは1体しか生成されないand護衛持ちモンスターの護衛は生成されない
というのはどうでしょう。

クリスタルキューブ1体なら純魔でも毒針で対処できる範囲ですし、
一般論として1vs1で倒す手段が絶無というモンスターは幾ら個性的でもアウトだと思われます。

単体召喚の脅威が下がるので複数召喚持ちに入れ替える等の調整が必要になるかもしれませんが、
破壊・抹殺が禁止されている@側と釣り合いを取るという意味ではこれでもいいかなと。

**Reply by daradarach** [reply]
つまり毒針をクエストに持ち込まない純魔は死ねということですね。

**Comment by dis-** [comment]
RES_ALL持ちをクエストに出現しない処理を実装しました。(Alpha23から反映されます)
その他の対処困難なモンスターの出現バランス調整は検証が済み次第実施します。
現在4.のレアリティ勾配を通常ダンジョンと変える案をベースにした対応を検討中です。
